components/server/src/api/workspace-service-api.ts

```diff
@@ -174,4 +182,60 @@ export class WorkspaceServiceAPI implements ServiceImpl<typeof WorkspaceServiceI
         response.workspace = this.apiConverter.toWorkspace(info);
         return response;
     }
+
+    async getWorkspaceDefaultImage(
+        req: GetWorkspaceDefaultImageRequest,
+        _: HandlerContext,
+    ): Promise<GetWorkspaceDefaultImageResponse> {
+        if (!req.workspaceId) {
+            throw new ApplicationError(ErrorCodes.BAD_REQUEST, "workspaceId is required");
+        }
+        const image = await this.workspaceService.getWorkspaceDefaultImage(ctxUserId(), req.workspaceId);
+        const response = new GetWorkspaceDefaultImageResponse({
+            defaultWorkspaceImage: image,
+        });
+        return response;
+    }
+
+    async sendHeartBeat(req: SendHeartBeatRequest, _: HandlerContext): Promise<SendHeartBeatResponse> {
+        if (!req.workspaceId) {
+            throw new ApplicationError(ErrorCodes.BAD_REQUEST, "workspaceId is required");
+        }
+        const info = await this.workspaceService.getWorkspace(ctxUserId(), req.workspaceId);
```

Comment:
why do we need it?

Reply:
See https://github.com/gitpod-io/gitpod/pull/19129#discussion_r1407281339 to make less changes in workspace-service and server-impl, it makes sense to do it here.

Reply:
ok, an issue to clean up after JSON-RPC is gone then 🙏